### ISSUE TITLE
Bug - sampleplayer with negative speed does not respect the loop option

### ISSUE THREAD

**Issue by PowerUser64** [BespokeSynth/BespokeSynth#1317]
## Describe the bug

When a sampleplayer has negative speed (so it plays backward), it does not respect the loop option. Instead, it loops in reverse endlessly.

## Build info

ff9397d12422d616c4174fb572db04f3e4583c92

## Steps To Reproduce

1. Get a sampleplayer with some audio in it
2. Set the speed parameter to -1
3. Press play with loop unchecked

The sample will loop endlessly in reverse.

## Expected vs. actual behavior

The sampleplayer should only loop when loop is enabled, even when playing in reverse.

**Comment by merkys** [comment]
I stumbled into the same issue at 63be9f82b749ec2d22a2e9e5b11d19efcb562f22. What is more, it seems that the position display in the GUI (the vertical bar moving across the sample) is not shown (stands still) for speeds < 0.

**Comment by PowerUser64** [comment]
Yup, I remember this - it stops whenever it reaches zero, even if it has looped once.

https://github.com/user-attachments/assets/3300249c-6d0b-4c24-854f-2cc4bf29f941

This recording also features some other interesting behavior, that the playhead stays at zero even after forward playback is resumed, staying there for the duration that it was playing in the reverse.

It seems like the problem is the playhead is only moved to the start after the internal playhead reaches the end of the recording. It should also be moved to the start when it reads negative sample positions.

**Comment by merkys** [comment]
Looking at the code, I see that the sample offset (`mOffset`) is a double which when played is not confined to be >= 0. Thus when it reaches 0 it continues to decrease playing the sample over and over again. I managed to confine it within 0 and sample length when loop option is not used, but then hitting play when `mOffset` is at 0 does not play any sound at all. I guess hitting play should move `mOffset` to the end of the sample and this should solve both the issues with stopping and the playhead movement.

**Comment by merkys** [comment]
I have opened PR #1891 resolving issues with negative speeds in `sampleplayer`. I would really appreciate some testing :smile:
